New test script and fixing of MacOSX problems

greatfireball opened the pull request:
I create a new test.sh which behaves now like a TAP test. Therefore, perl's `prove` might be used to run all tests. 

Moreover, I fixed the problems with `readlink -f` under MacOSX

Fixes #1 
Fixes #2 

<!---
@huboard:{"order":4.0,"milestone_order":4,"custom_state":""}
-->

greatfireball committed:
Added a bash function to avoid redundancy

I added a litte bash function to avoid redundancy during testing

greatfireball committed:
Removed first to fasta checks due to they are already performed using the test_fasta function

greatfireball committed:
Removed deletion of temporary files to start of test script

greatfireball committed:
Avoiding test problems if output files are not existing

greatfireball committed:
Renamed PIPE_FLAG to SPECIAL_FLAG

greatfireball committed:
Set an empty default SPECIAL_FLAG

greatfireball committed:
Added a test against individual files from last run controlled by SPECIAL_FLAG

greatfireball committed:
Added the third test based on the bash function

greatfireball committed:
Removed old test #3

greatfireball committed:
Added the fifth test based on the bash function

greatfireball committed:
Removed old test #5

greatfireball committed:
Added the sixth test based on the bash function

greatfireball committed:
Removed old test #6

greatfireball committed:
Added the fourth test based on the bash function plus one preparation test

greatfireball committed:
Removed old test #4

greatfireball committed:
Added a commentary to show where the test will be started

greatfireball committed:
Renamed test function to test_SeqChunker, corrected the test-names, and moved input file definition down to tests

greatfireball committed:
Added the first FASTQ test based on the bash function

greatfireball committed:
Removed old FASTQ test #1

greatfireball committed:
Added the second FASTQ test based on the bash function

greatfireball committed:
Removed old FASTQ test #2

greatfireball committed:
Added the third FASTQ test based on the bash function

greatfireball committed:
Removed old FASTQ test #3

greatfireball committed:
Added the fourth FASTQ test based on the bash function plus one preparational test

greatfireball committed:
Removed old FASTQ test #4

greatfireball committed:
Added the fifth FASTQ test based on the bash function plus one preparational test

greatfireball committed:
Removed old FASTQ test #5

greatfireball committed:
Cleaning up

greatfireball committed:
Skipping md5 based tests for now

greatfireball committed:
Now cycling through all implementations of SeqChunker

greatfireball committed:
Added TAP directive support and enable TODO directive for SeqChunker-sed, due to it seems to be broken

greatfireball committed:
New version 0.4 should work on MacOSX as well as Linux

greatfireball committed:
Put declaration of TAP_DIRECTIVE outside the current block

greatfireball committed:
Added a test, if bash >= 4 is present

review-ninja commented:
<a href="https://app.review.ninja/BioInf-Wuerzburg/SeqChunker/pull/4" target="_blank"><img src="https://app.review.ninja/25783014/pull/4/badge" alt="ReviewNinja"/>

greatfireball commented:
Approved

[![Approved with PullApprove](https://img.shields.io/badge/pullapprove-approved-brightgreen.svg)](https://pullapprove.com/BioInf-Wuerzburg/SeqChunker/pull-request/4/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=BioInf-Wuerzburg/SeqChunker)

greatfireball committed:
Fixed a typo in to test, which results in failing test without recognizing it

greatfireball committed:
Each test generates individual set of output files

Changed the way, the combined output will be generated. So each test will
generate its own set of output files. The name of the output files are generated
according to:

	  tmp.TESTNUMBER.*

So the tests if the correct output was generated can now avoid diff, but use the
md5sum of the output instead.

greatfireball committed:
Create .travis.yml

Travis-CI test should run from now on

greatfireball committed:
Added testing under osx and linux to travis.yml

I am using the new feature to test under different operation systems.

Therefore, we should be able to guarantee that the SeqChunker works under Linux and MacOSX

greatfireball committed:
Skipping installation step

According to https://docs.travis-ci.com/user/customizing-the-build/#Customizing-the-Installation-Step the installation step to resolve dependencies can be avoided. In our project that makes sense due to we have no real perl program running

greatfireball committed:
Ignoring build failures under osx

Due to we are sure about the MacOSX version, I tollerate build failures under osx

greatfireball committed:
Switch to local variables for function

greatfireball committed:
Added a new function for md5 based tests

greatfireball committed:
Added a reference.md5 file for expected MD5 sums for first 14 tests

greatfireball committed:
Removed old diff based test for expected output by md5 checksum based approach

greatfireball committed:
Changed to prove harnest to run tests

This is expected to fail for the original test script, but should work on the new one

greatfireball committed:
Added missing md5 reference sums

To generate the sums based on the original one, I used the following perl statement:

```
perl -ne '

     unless (/tmp/) { print; next }

     print;

     foreach my $fact (map {$_*14} (1..3))
     {
	$bla=$_;
	$bla=~s/(?<=\.)(\d+)(?=\.)/$1+$fact/eg; print $bla;
     }

' reference.md5
```

greatfireball committed:
Merge pull request #8 from BioInf-Wuerzburg/feature/travis-ci-testing

Testing at Travis-CI is now available

greatfireball committed:
Removed the allowed failure on MacOSX
